pkg/virt-controller/watch/vmi.go

```diff
@@ -863,10 +871,16 @@ func (c *VMIController) listVMIsMatchingDataVolume(namespace string, dataVolumeN
 	for _, obj := range objs {
 		vmi := obj.(*virtv1.VirtualMachineInstance)
 		for _, volume := range vmi.Spec.Volumes {
-			if volume.VolumeSource.DataVolume == nil {
```

Comment:
these changes don't seem correct to me.  The function used to return VMIs with a specific datavolume.  This will return VMIs with ANY non DataVolume volume plus matching datavolume volumes.

Reply:
No because if PVC == nil (so any non PVC volume types) it will check what it did before. So in short it is now returning all VMIs with a PVCs (regardless of if they have an owning DV) and all VMIs that have a DV

I do agree it is hard to understand what is actually going on. Let me rewrite it in a manner that makes more sense. I was trying to stay in the format that was there, but it is hard to understand what is going on when it can be written much more clearly.